File: back-end/src/main/java/com/huyphan/models/SearchCriteria.java
package com.huyphan.models;

import com.huyphan.models.enums.ConstructorName;
import com.huyphan.models.enums.SearchCriteriaName;
import lombok.Getter;
import lombok.Setter;

/** Searching criteria. */
@Getter
@Setter
public class SearchCriteria {

	/** Value used by this criteria to filter query results. */
	private Object value;

	/** Criteria name. */
	private SearchCriteriaName name;

	/** Construct which will use this search criteria to constructs Predicate. */
	private ConstructorName constructorName;
}

File: front-end/src/app/core/models/search-criteria.ts
import { SearchCriteriaName } from '../enums/search-criteria-name';
import { SearchCriteriaUsageDescription } from '../enums/search-criteria-usage-description';

/** Represents a search criteria used to search for an item. */
export interface SearchCriteria {
  /** Name of the search criteria. */
  readonly name: SearchCriteriaName;

  /** Value of this search criteria. */
  readonly value: Object;

  /** Description of what this search criteria used for. */
  readonly usageDescription: SearchCriteriaUsageDescription;
}
